pkg/apis/batch/validation/validation.go

```diff
@@ -389,9 +401,130 @@ func validatePodFailurePolicyRuleOnExitCodes(onExitCode *batch.PodFailurePolicyO
 	return allErrs
 }
 
+func validateSuccessPolicy(spec *batch.JobSpec, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+	criteriaPath := fldPath.Child("criteria")
+	if len(spec.SuccessPolicy.Criteria) == 0 {
```

Comment:
Similarly to PodFailurePolicy, we should also ensure we have a limit to how many success policy criteria can be specified. I don't expect too many to be specified, but it'd be good to put a cap here.

Reply:
That makes sense. I will add a limit the same as the podFailurePolicy.

Reply:
Done.